### PR TITLE
Changed Windows installation layout to fix cmake issues

### LINKED ISSUE #41: OpenCVConfig.cmake is broken on windows
If I build program with cmake, it can not find opencv.

```
-- OpenCV ARCH: x64
-- OpenCV RUNTIME: vc14
-- OpenCV STATIC: OFF
CMake Warning at D:/env/conda64_3.7/Library/OpenCVConfig.cmake:166 (message):
  Found OpenCV Windows Pack but it has no binaries compatible with your
  configuration.

  You should manually point CMake variable OpenCV_DIR to your build of OpenCV
  library.
Call Stack (most recent call first):
  CMakeLists.txt:85 (FIND_PACKAGE)


CMake Error at CMakeLists.txt:85 (FIND_PACKAGE):
  Found package configuration file:

    D:/env/conda64_3.7/Library/OpenCVConfig.cmake

  but it set OpenCV_FOUND to FALSE so package "OpenCV" is considered to be
  NOT FOUND.


-- Configuring incomplete, errors occurred!
```

``` cmake
# in OpenCVConfig.cmake
if(OpenCV_RUNTIME AND OpenCV_ARCH)
  if(OpenCV_STATIC AND EXISTS "${OpenCV_CONFIG_PATH}/${OpenCV_ARCH}/${OpenCV_RUNTIME}/staticlib/OpenCVConfig.cmake")
    if(OpenCV_CUDA AND EXISTS "${OpenCV_CONFIG_PATH}/gpu/${OpenCV_ARCH}/${OpenCV_RUNTIME}/staticlib/OpenCVConfig.cmake")
      set(OpenCV_LIB_PATH "${OpenCV_CONFIG_PATH}/gpu/${OpenCV_ARCH}/${OpenCV_RUNTIME}/staticlib")
    else()
      set(OpenCV_LIB_PATH "${OpenCV_CONFIG_PATH}/${OpenCV_ARCH}/${OpenCV_RUNTIME}/staticlib")
    endif()
  elseif(EXISTS "${OpenCV_CONFIG_PATH}/${OpenCV_ARCH}/${OpenCV_RUNTIME}/lib/OpenCVConfig.cmake")
    if(OpenCV_CUDA AND EXISTS "${OpenCV_CONFIG_PATH}/gpu/${OpenCV_ARCH}/${OpenCV_RUNTIME}/lib/OpenCVConfig.cmake")
      set(OpenCV_LIB_PATH "${OpenCV_CONFIG_PATH}/gpu/${OpenCV_ARCH}/${OpenCV_RUNTIME}/lib")
    else()
      set(OpenCV_LIB_PATH "${OpenCV_CONFIG_PATH}/${OpenCV_ARCH}/${OpenCV_RUNTIME}/lib")
    endif()
  endif()
endif()
```

In conda, the lib folder is not located at ${OpenCV_CONFIG_PATH}/${OpenCV_ARCH}/${OpenCV_RUNTIME}/lib.

### LINKED ISSUE #167: Windows build fails for a library which depends on OpenCV
I'm trying to build a recipe of [Aruco](https://github.com/conda-forge/aruco-feedstock), a C++ library which depends on OpenCV.
I've troubles for the compilation on Windows ([link of the CI log](https://ci.appveyor.com/project/conda-forge/aruco-feedstock/builds/25827938)).
```
  Found OpenCV Windows Pack but it has no binaries compatible with your
  configuration
```
It seems like the CI environment is not the same as the one it was built on. But, both are conda-smithy generated pipelines and use only x86-64 processors and Visual Studio 14 2015 as a compilator.
As found on other issues, I already explicitly specify to use a x64 compilation with` cmake -G "Visual Studio 14 2015 Win64"                                                                   `
Does someone already experienced that issue ?

### PR DESCRIPTION
**Pull request by mshabunin**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

resolves #167
resolves #41
replaces #112

Affected platform: _Windows_
Tuned build process to make it compatible with conda files layout. cmake files will be installed to the `cmake` subdirectory.

### CODE REVIEW

**Comment by show0k** [comment]
If this PR works, it would be useful to push it also in the 3.4.x branch :)

**Comment by mshabunin** [comment]
According to CI logs all files got installed as expected:
```
...
-- Installing: D:/bld/opencv_1563457925240/_h_env/Library/cmake/x64/vc14/lib/OpenCVModules.cmake
-- Installing: D:/bld/opencv_1563457925240/_h_env/Library/cmake/x64/vc14/lib/OpenCVModules-release.cmake
-- Installing: D:/bld/opencv_1563457925240/_h_env/Library/cmake/x64/vc14/lib/OpenCVConfig-version.cmake
-- Installing: D:/bld/opencv_1563457925240/_h_env/Library/cmake/x64/vc14/lib/OpenCVConfig.cmake
-- Installing: D:/bld/opencv_1563457925240/_h_env/Library/cmake/OpenCVConfig-version.cmake
-- Installing: D:/bld/opencv_1563457925240/_h_env/Library/cmake/OpenCVConfig.cmake
-- Installing: D:/bld/opencv_1563457925240/_h_env/Library/lib/opencv_core410.lib
-- Installing: D:/bld/opencv_1563457925240/_h_env/Library/bin/opencv_core410.dll
...
```
I'm not sure how we can test this patch... It can be simple application using cmake for build. I'll try to implement it.

**Comment by hmaarrfk** [comment]
> If this PR works, it would be useful to push it also in the 3.4.x branch !

Has anybody requested anything to happen for the 3.4 branch? There are quite a few PRs that didn't get backported (mostly @mshabunin 's work). No user has requested anything from 3.4 to my knowledge.

Are you making this request as a "user of 3.4" or a concerned party for a "would be user"?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -142,6 +143,10 @@ test:
         - test -f $PREFIX/lib/libopencv_freetype${SHLIB_EXT}     # [unix]
         - python -c "import cv2; assert 'Unknown' not in cv2.videoio_registry.getBackendName(cv2.CAP_V4L)"  # [linux]
         - python -c "import cv2, re; assert re.search('Lapack:\s+YES', cv2.getBuildInformation())"
+        - mkdir -p cmake_build_test && pushd cmake_build_test
+        - cmake -G "${CMAKE_GENERATOR}" -D "OpenCV_DIR=${PREFIX}/cmake" "${RECIPE_DIR}"
```
Comment (hmaarrfk) [inline recipe/meta.yaml]: there is a way to specify a test file that should be copied into the testing stage. Not sure if that is what is giving you trouble. You are really being thorough :D.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -142,6 +146,10 @@ test:
         - test -f $PREFIX/lib/libopencv_freetype${SHLIB_EXT}     # [unix]
         - python -c "import cv2; assert 'Unknown' not in cv2.videoio_registry.getBackendName(cv2.CAP_V4L)"  # [linux]
         - python -c "import cv2, re; assert re.search('Lapack:\s+YES', cv2.getBuildInformation())"
+        - mkdir -p cmake_build_test && pushd cmake_build_test
+        - cmake -G "Ninja" -D "OpenCV_DIR={{ PREFIX }}/cmake" ..
```
Comment (hmaarrfk) [inline recipe/meta.yaml]: why do we need this? don't we want cmake to be able to find this automatically? without the need for any kinds of defines from the user?

Comment (mshabunin) [reply recipe/meta.yaml]: Thanks for fixes!
Yes, I'll remove it.

**Comment by hmaarrfk** [comment]
Wow, so this is working now? We got this down to a configuration issue instead of a patch?

@jakirkham did you want to review?

**Comment by jakirkham** [comment]
Seems reasonable. Happy to trust your judgement here 🙂

**Comment by hmaarrfk** [comment]
@show0k I'm just following up with your request to backport to 3.4. Did you still feel it was necessary?

**Comment by show0k** [comment]
Sorry, I didn't see the notification.
I was requesting the backport as a user of 3.4. I'm using righ now "libopencv" package as a workaround for 3.4, and it would be more conveniant to have the same dependancy for 3.4 and for 4.1.

**Comment by hmaarrfk** [comment]
Ok that's totally fair

**Pull request merged**